### PR TITLE
add graphql issue fetch example

### PR DESCRIPTION
**Pull request by BenTheElder**
This tool isn't necessarily intended to be used at all, but it does provide a reference for getting all issues with the GitHub v4 GraphQL API, which may be useful as we move away from mungegithub.

### CODE REVIEW

File: experiment/graphql_issue_example.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            owner="kubernetes"
+    name: this GitHub repo name as in github.com.kubernetes/test-infra ->
+            name = "test-infra"
+    token: a GitHub API token string
+    after: this should be None or the endCursor from pageInfo
+
+    Returns:
+    A requests.Response object
+    """
+    query = ("query($owner:String!, $name:String!, $after:String) {\n"
```
Comment (rmmh) [inline experiment/graphql_issue_example.py]: Can you change this to a """ string for clarity here?

File: experiment/graphql_issue_example.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        for entry in issues:
+            issue_func(entry)
+        page_info = data["repository"]["issues"]["pageInfo"]
+        if not page_info["hasNextPage"]:
+            break
+        response = get_issues(owner, name, token, page_info["endCursor"])
+
+
+def main():
+    token = sys.argv[-1]
```
Comment (rmmh) [inline experiment/graphql_issue_example.py]: Use argparse:

```python
parser = argparse.ArgumentParser()
parser.add_option('--org', default='kubernetes')
parser.add_option('--repo', default='test-infra')
parser.add_option('token', help='GitHub auth token.')
options = parser.parse_args()
print("getting issues for: %s/%s" % (options.org, options.repo)
```

File: experiment/graphql_issue_example.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    org = "kubernetes"
+    repo = "test-infra"
+    print("getting issues for: %s/%s" % (org, repo))
+    # TODO: replace this with with something more useful?
+    def issue_func(issue):
+        print(issue)
+    get_all_issues(org, repo, token, issue_func)
```
Comment (rmmh) [inline experiment/graphql_issue_example.py]: `show_progress=True`?

Comment (BenTheElder) [reply experiment/graphql_issue_example.py]: It's just printing the issues now so showing progress seemed redundant. we could dump these to a CSV or something but they have nested json. I don't actually have a good use case for this script besides showing how to iterate the issues with graphql.

File: experiment/graphql_issue_example.py (hunk 2 above)
Comment (BenTheElder) [reply experiment/graphql_issue_example.py]: done.

File: experiment/graphql_issue_example.py (hunk 1 above)
Comment (BenTheElder) [reply experiment/graphql_issue_example.py]: done.

**Pull request merged**